internal/olm/client/client.go

```diff
@@ -214,7 +216,70 @@ func (c Client) DoCSVWait(ctx context.Context, key types.NamespacedName) error {
 		return curPhase == olmapiv1alpha1.CSVPhaseSucceeded, nil
 	}
 
-	return wait.PollImmediateUntil(time.Second, csvPhaseSucceeded, ctx.Done())
+	err := wait.PollImmediateUntil(time.Second, csvPhaseSucceeded, ctx.Done())
+	if err != nil && errors.Is(err, context.DeadlineExceeded) {
```

Comment:
Coming back to this from a prior review: why check deployments/pods only when a timeout occurs? Shouldn't we print deployment/pod status when any error occurs? We can ignore "not found" errors in case something happened before the operator deployment was created.

Reply:
@joelanford WDYT? I know you have mentioned to check only for deadline exceeded situation.